### PR TITLE
simplify address handling by storing the internal dns name in the cluster object

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Simplify address handling by storing the internal dns name in the cluster object

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -138,8 +138,12 @@ type NetworkRanges struct {
 type ClusterAddress struct {
 	// URL under which the Apiserver is available
 	URL string `json:"url"`
+	// Port is the port the API server listens on
+	Port int32 `json:"port"`
 	// ExternalName is the DNS name for this cluster
 	ExternalName string `json:"externalName"`
+	// InternalName is seed cluster internal absolute DNS name the API server
```
Comment (alvaroaleman) [inline api/pkg/crd/kubermatic/v1/cluster.go]: The sentence lacks a `to`

File: api/pkg/resources/address/address.go [hunk 2]
```diff
@@ -51,26 +51,42 @@ func SyncClusterAddress(ctx context.Context,
...
+	url := fmt.Sprintf("https://%s:%d", externalName, nodePort)
 	if cluster.Address.URL != url {
 		modifiers = append(modifiers, func(c *kubermaticv1.Cluster) {
 			c.Address.URL = url
 		})
-		glog.V(4).Infof("Set URL for cluster %s to '%s'", cluster.Name, url)
+		glog.V(2).Infof("Set URL for cluster %s to '%s'", cluster.Name, url)
+	}
+
+	internalName := fmt.Sprintf("%s.%s.svc.cluster.local", resources.ApiserverExternalServiceName, cluster.Status.NamespaceName)
```
Comment (alvaroaleman) [inline api/pkg/resources/address/address.go]: Maybe add a trailing dot to the DNS name?

Comment (mrIncompetent) [reply api/pkg/resources/address/address.go]: Done with https://github.com/kubermatic/kubermatic/pull/3195/commits/3c78761972709351eb7b3ee194349a6cfe857f5e

File: api/pkg/crd/kubermatic/v1/cluster.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/crd/kubermatic/v1/cluster.go]: Done with https://github.com/kubermatic/kubermatic/pull/3195/commits/3c78761972709351eb7b3ee194349a6cfe857f5e

**Pull request merged**